app/models/message_summarizer/problem_order_details_summary.rb

```diff
@@ -0,0 +1,19 @@
+class MessageSummarizer::ProblemOrderDetailsSummary < MessageSummarizer::MessageSummary
+  private
+
+  def allowed?
+    facility && ability.can?(:show_problems, Order)
+  end
+
+  def get_count
+    facility.problem_non_reservation_order_details.count
+  end
+
+  def l18n_key
+    "message_summarizer.problem_order_details"
```

Comment:
Since they're all under `message_summarizer`, would it make sense to pull that up into the superclass?